src/util/postgres_client.js

```diff
@@ -149,13 +153,46 @@ function handle_ops_encoding(schema, val) {
 
 let query_counter = 0;
 
+async function log_query(pg_client, q, tag, millitook, pool_count, should_explain) {
+    const explain_q = { text: 'explain ' + q.text, values: q.values };
+    let explain_rows = ['explain:\tno explain'];
+    try {
+        if (should_explain && process.env.EXPLAIN_QUERIES === 'true') {
+            let explain_res = await pg_client.query(explain_q);
+            explain_rows = explain_res.rows.map(row => 'explain:\t' + row['QUERY PLAN']);
+        }
+    } catch (err) {
+        console.error('got error on explain', explain_q, err);
+        explain_rows = ['explain:\texplain failed'];
+    }
+
+    let stack = (new Error()).stack.split('\n').slice(3).join(',');
+    const msgs = [];
+    msgs.push(`stack: ${stack}`);
+    msgs.push(`tag: ${tag}`);
+    msgs.push(`query: ${q.text}`);
+    msgs.push(`took: ${millitook}`);
+    msgs.push(`pool: ${pool_count}`);
+    msgs.push(`explain:`);
+    let log_message = msgs.concat(explain_rows).join('\nQUERY_LOG:\t');
+    console.log(`\nQUERY_LOG:\t${log_message}`);
```

Comment:
There seems to be q lot of objects being created here just for logging.
I would try to keep this more straightforward code that prints the string.
Also I think it would be best to keep the format as a json object and not a multiline log print.